### DISCUSSION TITLE
Is it possible to configure wildcard in the middle of a path

### DISCUSSION THREAD

**Discussion by suntsu42** [dotnet/yarp#2067] [Q&A] [answered]
The following example works, and applies to all content after /foo
```json
  "route1": {
        "ClusterId": "cluster1",
        "AuthorizationPolicy": "customPolicy",
        "Match": {
          "Path": "/foo/{**catch-all}"
        }
      },
```

I now want to add a route which applies for all traffic which ends with something specific, but the middle part should be wildcard
```json
  "route1": {
        "ClusterId": "cluster1",
        "AuthorizationPolicy": "customPolicy",
        "Match": {
          "Path": "/foo/{**wildcard}/bar"
        }
      },
```
In this case, the error message is: A catch-all parameter can only appear as the last segment of the route template.

Is there a way to configure a route which does apply a wildcard which is in the middle of the uri? 
It my example, it should be applied to 
/foo/demo/bar
/foo/xyz/bar

**Comment by suntsu42** [comment, marked as answer]
I think i now found the solution by using message:regex 
```
          "Path": "/foo/{message:regex(.*)}/bar/{**catch-all}"
```

**Reply by personwithproxy** [reply]
But transforms using the path from the regex (something like that:)
"Match": { "Path": "/foo/{message:regex(.*)}/bar/{**catch-all}" },
        "Transforms": [
          { "PathPattern": "/foo/$1/bar/{**catch-all}" }
        ]
are not possible, as far as I know. Or is there a way to do that and I just missed the point?

**Reply by Tratcher** [reply]
Try `{ "PathPattern": "/foo/{message}/bar/{**catch-all}" }`
